### ISSUE TITLE
object_camel_case_linter() flags methods from external packages

### ISSUE THREAD

**Issue by khughitt** [jimhester/lintr#392]
Currently, the `object_camel_case_linter` linter complains when there are variables / function names that are not in `snake_case`, even for methods defined in outside packages that the user has no control over.

For example, the following call to `caret::trainControl()` results in lintr warning:

```r
train_control <- trainControl(method = "cv", number = 10, savePred = TRUE, classProb = FALSE)

# warning| Variable and function names should be all lowercase. 
```

Would it be possible, perhaps, to check if a function is being called without having been defined in the file, and if so, assume that it is an external method and ignore both the function name / arguments?

Even if it _is_ a user method that has been sourced, lintr will still catch the issue in the file that is being sourced.

**Comment by russHyde** [comment]
The `camel_case_linter` has been added to the deprecated functions.

If you want to lint against object names (and you are using the development version of lintr), you could use `object_name_linter(styles = c("my_set_of", "allowable_styles"))`. The object_name_linter now flags variables and formal arguments at the point where they are defined, rather than where they are used.

**Comment by khughitt** [comment]
Great! Thanks for the update / suggestion. Closing the issue since it's no longer relevant.
